profiles/schema.py

```diff
@@ -589,6 +601,26 @@ class Meta:
         "privileges to access this information.",
     )
 
+    def resolve_login_methods(self: Profile, info, **kwargs):
+        if info.context.user != self.user:
+            raise PermissionDenied(
+                "No permission to read login methods of another user."
+            )
+
+        amr = set(force_list(info.context.user_auth.data.get("amr")))
+
+        # For future software archeologists:
+        # This field was added to the API to support the front-end's need to know
+        # which login methods the user has used. It's only needed for profiles
+        # with helsinki-tunnus or Suomi.fi, so for other cases, save a couple
+        # API calls and return an empty list. There's no other reasoning for the
+        # logic here.
+        # Can remove this after Tunnistamo is no longer in use. Related ticket: HP-2495
+        if amr.intersection({"helsinki_tunnus", "heltunnistussuomifi", "suomi_fi"}):
+            return get_user_login_methods(self.user.uuid)
```

Comment:
Here (or in the function) we could check that the values which are returned are part of the enum. So only values which are part of the enum are returned and maybe add a `logger.warning(f"Loging methods found which are not part of the enum: {values_not_in_enum}")` for the values missing from the enum.

Tried this locally and since my account also had `recovery-authn-codes` I got the following error straight away:

```
 "Enum 'LoginMethodType' cannot represent value: 'recovery-authn-codes'",
```